src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -72,7 +72,7 @@
     public JsonParser(TypeToken<T> type) {
         this.type = type;
         this.gson = new GsonBuilder()
-                .registerTypeAdapter(Date.class, new SaltStackDateDeserializer())
+                .registerTypeAdapter(Date.class, new DateAdapter().nullSafe())
```

Comment:
Ok, fine with me. I will then just merge the patch as it is and open an issue about handling optional fields.